test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -420,35 +420,70 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 		})
 	})
 
-	// TODO: we have a dependency between tests(that in general bad practice, but saves us some tests run time),
-	// once we will want to run tests in the random order or without failFast we will need to refactor tests
 	Context("Updating of nodeSelector parameter and node labels", func() {
 		var mcp *machineconfigv1.MachineConfigPool
 		var newCnfNode *corev1.Node
 
 		newRole := "worker-test"
 		newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+		var labelsDeletion = false
 		newNodeSelector := map[string]string{newLabel: ""}
 
-		testutils.BeforeAll(func() {
+		//fetch the latest profile
+		profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		oldMcpSelector := make(map[string]string)
+
+		//fetch existing MCP Selector if exists in profile
+		if profile.Spec.MachineConfigPoolSelector != nil {
+			oldMcpSelector = profile.Spec.DeepCopy().MachineConfigPoolSelector
+		}
```

Comment:
we should have only variable declarations in the `Context` block. Trivial initializations (using constants, for examples) are OK, but more complex code, like getting the profile, should be moved in `BeforeEach` - or everywhere else suitable.